src/test/scala/org/graphframes/lib/ParallelPersonalizedPageRankSuite.scala

```diff
@@ -0,0 +1,115 @@
+ * limitations under the License.
+ */
+
+package org.graphframes.lib
+
+import org.apache.spark.sql.functions.col
+import org.apache.spark.sql.types.DataTypes
+import org.apache.spark.sql.Row
+import org.apache.spark.ml.linalg.{SQLDataTypes, SparseVector}
+
+import org.graphframes.examples.Graphs
+import org.graphframes.{GraphFrameTestSparkContext, SparkFunSuite, TestUtils}
+
+class ParallelPersonalizedPageRankSuite extends SparkFunSuite with GraphFrameTestSparkContext {
+
+  val n = 100
+
+  private def isLaterVersion(minVer: String): Boolean = {
+    val (minVersionMajor, minVersionMinor) = majorMinorVersion(minVer)
+    if (sparkVersionMajor != minVersionMajor) {
+      return sparkVersionMajor > minVersionMajor
+    } else {
+      return sparkVersionMinor >= minVersionMinor
+    }
+  }
+
+  test("Illegal function call argument setting") {
+    val g = Graphs.star(n)
+    val vertexIds: Array[Any] = Array(1L, 2L, 3L)
+
+    def checkPageRankRun(prc: ParallelPersonalizedPageRank): Unit = {
+      if (isLaterVersion("2.1")) {
+        intercept[IllegalArgumentException] { prc.run() }
+      } else {
+        intercept[NotImplementedError] { prc.run() }
+      }
+    }
+
+    lazy val prcSansNumIter = g.parallelPersonalizedPageRank
+      .sources(vertexIds)
+    checkPageRankRun(prcSansNumIter)
+
+    lazy val prcSansVertexIds = g.parallelPersonalizedPageRank
+      .numIter(15)
+    checkPageRankRun(prcSansVertexIds)
+  }
+
+  test("Star example parallel personalized PageRank") {
+    val g = Graphs.star(n)
+    val resetProb = 0.15
+    val numIter = 10
+    val vertexIds: Array[Any] = Array(1L, 2L, 3L)
+
+    lazy val prc = g.parallelPersonalizedPageRank
+      .numIter(numIter)
+      .sources(vertexIds)
+      .resetProbability(resetProb)
+
+    if (isLaterVersion("2.1")) {
+      val pr = prc.run()
+      TestUtils.testSchemaInvariants(g, pr)
+      TestUtils.checkColumnType(pr.vertices.schema, "pagerank", SQLDataTypes.VectorType)
+      TestUtils.checkColumnType(pr.edges.schema, "weight", DataTypes.DoubleType)
+    } else {
+      intercept[NotImplementedError] { prc.run() }
+    }
+  }
+
+  test("friends graph with parallel personalized PageRank") {
+    val g = Graphs.friends
+    val resetProb = 0.15
+    val numIter = 10
+    val vertexIds: Array[Any] = Array("a")
+    lazy val prc = g.parallelPersonalizedPageRank
+      .numIter(numIter)
+      .sources(vertexIds)
+      .resetProbability(resetProb)
+
+    if (isLaterVersion("2.1")) {
+      val pr = prc.run()
+      val prInvalid = pr.vertices
+        .select("pagerank")
+        .collect()
+        .filter { row: Row =>
+          vertexIds.size != row.getAs[SparseVector](0).size
+        }
+      assert(prInvalid.size === 0,
+        s"found ${prInvalid.size} entries with invalid number of returned personalized pagerank vector")
+
+      val gRank = pr.vertices
+        .filter(col("id") === "g")
+        .select("pagerank")
+        .first().get(0).asInstanceOf[SparseVector]
```

Comment:
Thank you! 
Did a grep `find src -name '*.scala' | xargs grep -n "asInstanceOf"` 
It seems that all the none `getAs[T]` use cases are replaced. 
I will keep this in mind for later work.